StylableSwiftUI/Classes/Stylist.swift

```diff
@@ -26,11 +26,19 @@ public struct Style {
 @available(iOS 13.0.0, *)
 public class Stylist: ObservableObject {
 
+    public enum Mode {
+        case atomicDesign
+        case nds
```

Comment:
Shouldn’t it be natural (for natural design system) and atomic?

Reply:
If the only difference is how we want to select a matching style, why not just pass in a function instead of an enum i.e. 

```swift
typealias StylistMatchingMode = 
getFirstMatchingSyle(for identifier: StylistIdentifier) -> Style?
self.styles.first(where: { $0.identifier.matches(identifier) })

Reply:
Yeah that's a good idea. There is also the question of preventing the sorting happening at the beginning though?

Reply:
Oh blast, that's true - I'd forgotten about sorting! Hang on . . . .

Reply:
OK, I think I've built in lots of assumptions about how the identifier will match into the identifiers themselves i.e. specificity only applies to Atomic Design, and I don't want to think about how changing that will affect identifiers being `Comparable`!

So, what about this (and it's this post where you might want to accuse me of over-engineering :)

```swift

/// A _real_ type, not a `typealias`
protocol StylistMatchingMode {
  func insert(identifiers: [StylistIdentifier], into: [StylistIdentifier]) -> [StylistIdentifier]
  func firstMatch(styles: [Style], toIdentifier: StylistIdentifier) -> Style?
}

struct AtomicDesign: StylistMatchingMode {

  func insert(identifiers: [StylistIdentifier], into existing: [StylistIdentifier]) -> [StylistIdentifier] {
    var new = existing
    new.append(contentsOf: identifiers)
    return new.sorted { blah blah blah }
  }

  func firstMatch(styles: [Style], toIdentifier: StylistIdentifier) -> Style? {
    styles.first { $0.identifier.matches(identifier) }
  }
}
```

You get the idea. All the matching specific stuff would live inside `StylistMatchingMode`, and the Stylist would just defer to that.

We should then probably move everything which is specific to a matching mode out from `StylistIdentifier` i.e. score, specificity etc - they could probably still live in extensions, but those extensions should be near the mode they are used in, and not just generally available.

Another question: how does this new matching system affect finding assets i.e. image names?

Reply:
@deanWombourne regular images work but images with variants are currently a WIP

Reply:
@deanWombourne / @kerrmarin are either of you using the atomicDesign mode in any apps? So far I'm not. If neither of you are, I suggest we drop atomicDesign for KISS, ease of future maintenance and ease of documentation reasons.